request-manager/graph/resolver.go

```diff
@@ -121,193 +123,168 @@ func (o *resolver) RequestArgs(jobArgs map[string]interface{}) ([]proto.RequestA
 	jobArgs := cfg.jobArgs
 
-	/* Add optional and static sequence arguments to map. */
-	seq, ok := o.AllSequences[sequenceName]
-	if !ok {
-		return nil, fmt.Errorf("cannot find definition for sequence: %s", sequenceName)
-	}
-	for _, arg := range seq.Args.Optional {
-		if _, ok := jobArgs[*arg.Name]; !ok {
-			jobArgs[*arg.Name] = *arg.Default
-		}
-	}
-	for _, arg := range seq.Args.Static {
-		if _, ok := jobArgs[*arg.Name]; !ok {
-			jobArgs[*arg.Name] = *arg.Default
-		}
-	}
-
-	/* Build request graph based on sequence graph. */
-	seqGraph, ok := o.SequenceGraphs[sequenceName]
+	// Build request graph based on sequence graph. We use the sequence graph
+	// as a template, traversing it in topological order and processing each
+	// of its nodes depending on what category it is (job, sequence, conditional).
+	// This sequence graph won't be mutated at all.
+	seqGraph, ok := r.sequenceGraphs[sequenceName]
 	if !ok {
+		// Graph checks should prevent this from happening.
+		// If this error is thrown, there's a bug in the code.
 		return nil, fmt.Errorf("cannot find sequence graph for sequence: %s", sequenceName)
 	}
 
-	reqGraph, err := o.newReqGraph(cfg.wrapperName, jobArgs)
+	// This is the graph we'll actually be filling out. It starts off empty,
+	// and we'll add nodes as we traverse the sequence graph.
+	reqGraph, err := r.newReqGraph(cfg.graphName, jobArgs)
 	if err != nil {
 		return nil, err
 	}
 
-	// Traverse sequence graph in topological order. Each sequence graph node corresponds to either a single
-	// job node, or an entire subgraph of nodes. Build the subgraph and insert it into the larger sequence
-	// graph.
+	// Traverse sequence graph in topological order. Each sequence graph node
+	// corresponds to either a single job node, or an entire subgraph of nodes.
+	// Build the subgraph and insert it into the larger sequence graph.
 	idMap := map[string]*Graph{} // sequence graph node id --> corresponding subgraph
 	for _, seqGraphNode := range seqGraph.Order {
-		n := seqGraphNode.Spec
-
-		// Find out how many times this node has to be repeated
-		elements, lists, err := o.getIterators(n, jobArgs)
+		nodeSpec := seqGraphNode.Spec
+
+		// 1. Do sequence expansion.
+		// Find out how many times this node has to be repeated.
+		// If this node has a `each:` field, then lists[i] and
+		// elements[i] describe the ith `list:element`. Specifically,
+		// lists[i] is the actual list value, not just the name of the
+		// job arg.
+		elements, lists, err := r.getIterators(nodeSpec, jobArgs)
 		if err != nil {
-			return nil, fmt.Errorf("in seq %s, node %s: invalid 'each:' %s", sequenceName, n.Name, err)
+			return nil, fmt.Errorf("in seq %s, node %s: invalid 'each:' %s", sequenceName, nodeSpec.Name, err)
 		}
 
-		// All the graphs that make up this sequence graph node's subgraph
+		// All the graphs that make up this sequence graph node's subgraph,
+		// one for each time the node is repeated
 		components := []*Graph{}
```

Comment:
Now that code is more clear, I think we'll enhance that clarify even further by getting ride of this very generic term that only appears here, once, and nowhere else. I think `seqExpansion` (or maybe `expandedSeq`) is better because that's it's primary purpose and that var name is immediately intention-revealing to FE who know basic Spin Cycle design elements.

And to really make the life of the FE easier, let's give an example:
```
// For example, with 3 sequences like S1->S2->S3, if we're currently
// in S1 (i.e. reqGraph=S1) and S2 is expanded ("each: foo:bar"),
// we'll build N-many parallel S2 and save them in seqExpansion.
// Then, after this `for` loop, we'll wrap all seqExpansion graphs
// in a new graph so they have a single source and sink node.
```